bodhi-client/bodhi/client/bindings.py

```diff
@@ -245,7 +248,14 @@ def ensure_auth(self):
                     break
                 if resp.status_code == 401:
                     self.clear_auth()
-                    self.oidc.login()
+                    # by default, when login_with_kerberos is True, we use Kerberos to authenticate
+                    # and if that fails, fall back to the interactive browser-based OIDC login
+                    try:
+                        if self.login_with_kerberos:
+                            self.oidc.login_with_kerberos()
+                    except OIDCClientError as e:
+                        log.info("Failed to login with Kerberos: %s", e)
+                        self.oidc.login()
```

Comment:
Please let me know if this is what you had in mind. If so, I'm going to cover this by a test case and resolve the 2 test failures.